modules/ocha_ai_chat/src/Form/OchaAiChatChatForm.php

```diff
@@ -363,6 +376,11 @@ public function buildForm(array $form, FormStateInterface $form_state, ?bool $po
       'disable-refocus' => TRUE,
     ];
 
+    if ($this->moduleHandler->moduleExists('honeypot')) {
+      $honeypot_options = ['honeypot', 'time_restriction'];
+      honeypot_add_form_protection($form, $form_state, $honeypot_options);
```

Comment:
This function is deprecated: https://git.drupalcode.org/project/honeypot/-/blob/2.1.x/honeypot.module?ref_type=heads#L129

This should be replaced by `\Drupal::service('honeypot')->addFormProtection($form, $form_state, $honeypot_options);`.

However, PHPCS will complain because we are not injecting the `honeypot` service (which may not be defined).

 One option could be to add `hook_form_FORM_ID_alter()` implementation in `ocha_ai_chat.module` for the chat form, check the `honeypot` module and/or service exists and then call the `addFormProtection` method of the service.

Reply:
How about adding Honeypot as a dependency? It's a bit of a dramatic response to a deprecated function - but is there a case where we might offer chat without honeypot protection?